### PR TITLE
add allen-brain user story by @jsiegle

### PR DESCRIPTION
**Pull request by jhamman**
towards: https://github.com/xarray-contrib/xarray.dev/issues/344

cc @jsiegle

### CODE REVIEW

**Comment by jhamman** [comment]
@andersy005 - any ideas why this is failing to build?

File: src/posts/xarray-for-neurophysiology/index.md [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+from allensdk.brain_observatory.ecephys.ecephys_project_cache import EcephysProjectCache
+
+cache = EcephysProjectCache.from_warehouse(manifest=manifest_path)
+
+session = cache.get_session_data(819701982) # access data by session ID
+```
+
+The primary data for this session consists of spike times from 585 neurons (also called "units") simultaneously recorded across more than a dozen brain regions. This is what a 10-second snippet of the recording looks likes, with each dot representing one spike:
+
+<!-- TODO: add figure 1 -->
```
Comment (andersy005) [inline src/posts/xarray-for-neurophysiology/index.md:30]: @jhamman, it turns out that MDX doesn't support HTML style comments: https://github.com/mdx-js/mdx/issues/1042

the following solution should be sufficient to address the build failures.

```suggestion
{/* TODO: add figure 1 */}
```

File: src/posts/xarray-for-neurophysiology/index.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+
+unit_id = units[(units.probe_description == 'probeA')
+            	& (units.ecephys_structure_acronym == 'CA1')].index.values[26]
+nearby_lfp = lfp.sel(channel=units.loc[unit_id].peak_channel_id, method='nearest')
+
+b,a = butter(3, np.array([5, 9])/(1250 / 2), btype='bandpass')
+lfp_filt = filtfilt(b, a, hippocampal_lfp)
+lfp_phase = xr.DataArray(np.angle(hilbert(lfp_filt)), coords={'time': nearby_lfp.time})
+
+spike_phase = lfp_phase.sel(time=session.spike_times[unit_id], method='nearest')
```
Comment (andersy005) [inline src/posts/xarray-for-neurophysiology/index.md:137]:
```suggestion
from scipy.signal import butter, filtfilt, hilbert

unit_id = units[
    (units.probe_description == "probeA") & (units.ecephys_structure_acronym == "CA1")
].index.values[26]
nearby_lfp = lfp.sel(channel=units.loc[unit_id].peak_channel_id, method="nearest")

b, a = butter(3, np.array([5, 9]) / (1250 / 2), btype="bandpass")
lfp_filt = filtfilt(b, a, hippocampal_lfp)
lfp_phase = xr.DataArray(np.angle(hilbert(lfp_filt)), coords={"time": nearby_lfp.time})

spike_phase = lfp_phase.sel(time=session.spike_times[unit_id], method="nearest")
```

File: src/posts/xarray-for-neurophysiology/index.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+Special thanks to Nile Graddis for introducing me to Xarray and writing the AllenSDK methods described here. Thanks to Joe Hamman for providing feedback on this post.
+
+### References
+
+1. Stevenson & Körding (2011) _Nat Neurosci_ **14**: 139-142.
+2. Jun et al. (2017) _Nature_ **551**: 232-236.
+3. Demas et al. (2021) _Nat Methods_ **18**: 1103-1111.
+4. Rübel et al. (2022) _eLife_ **11**: e78362.
+5. De Vries et al. (2023) _arXiv_ 2212.08638
```
Comment (andersy005) [inline src/posts/xarray-for-neurophysiology/index.md:160]:
```suggestion
1. [Stevenson & Körding (2011) _Nat Neurosci_ **14**: 139-142](https://www.nature.com/articles/nn.2731).
2. [Jun et al. (2017) _Nature_ **551**: 232-236](https://www.nature.com/articles/nature24636).
3. [Demas et al. (2021) _Nat Methods_ **18**: 1103-1111](https://www.nature.com/articles/s41592-021-01239-8).
4. [Rübel et al. (2022) _eLife_ **11**: e78362](https://elifesciences.org/articles/78362).
5. [De Vries et al. (2023) _arXiv_ 2212.08638](https://arxiv.org/abs/2212.08638).
```

**Review by andersy005 — COMMENTED** [review]
💬 @jhamman & @jsiegle, thank you for this addition! i left some comments/suggestions (mostly code formatting)

**Comment by jsiegle** [comment]
Here are some edits that address @TomNicholas' suggestions: https://github.com/jsiegle/xarray.dev/commit/b5f5ee0753ecaf19754d8b3adfae1770d5afb301

**Comment by jhamman** [comment]
Thanks again for writing this up @jsiegle! It's live here: https://xarray.dev/blog/xarray-for-neurophysiology

**Pull request merged**